autode/utils.py

```diff
@@ -399,36 +441,96 @@ def wrapped_function(*args, **kwargs):
     return wrapped_function
 
 
-def timeout(seconds: float, return_value: Optional[Any] = None) -> Any:
+def _cleanup_after_exp_timeout_win() -> None:
     """
-    Function decorator that times-out after a number of seconds
+    If experimental timeout has been used, the ProcessPool has to be shutdown
+    otherwise File permission errors will be caused on Windows
+    """
+    if Config.use_experimental_timeout:
+        pool = loky.get_reusable_executor()
+        pool.shutdown()
 
-    ---------------------------------------------------------------------------
-    Arguments:
-        seconds:
 
+def _timeout_experimental(
+    seconds: float, return_value: Optional[Any] = None
+) -> Any:
+    """
+    Function decorator that times-out after a number of seconds, if
+    Config.use_experimental_timeout = True, otherwise no timeout
+    (experimental version, works on Windows)
+
+    Args:
+        seconds: The number of seconds to timeout
+        return_value: Value returned if the function times out
+
+    Returns:
+        (Any): result of function | return_value
+    """
+
+    def func_runner(connector, func, args, kwargs):
+        connector.send(os.getpid())
+        return func(*args, **kwargs)
+
+    def decorator(func):
+        def wrapper(*args, **kwargs):
+            # if user does not want it, there is no timeout
+            if not Config.use_experimental_timeout:
+                return func(*args, **kwargs)
+
+            pool = loky.get_reusable_executor(max_workers=1)
+            conn1, conn2 = multiprocessing.Pipe()
+            job = pool.submit(func_runner, conn1, func, args, kwargs)
+            job_pid = conn2.recv()
+            try:
+                res = job.result(timeout=seconds)
+                return res
+            except loky.TimeoutError:
+                if os.getpid() != job_pid:
+                    os.kill(job_pid, signal.SIGTERM)
+                # allow the loky process pool manager thread to wakeup
+                sleep(0.6)
```

Comment:
this is a scary magic number. what if someones on an old/slow machine that takes longer than this? is there a condition that can be checked?